umodbus/serial.py

```diff
@@ -104,20 +110,27 @@ def __init__(self,
                           # timeout_chars=2,  # WiPy only
                           # pins=pins         # WiPy only
                           tx=pins[0],
-                          rx=pins[1]
-                          )
+                          rx=pins[1])
 
         if ctrl_pin is not None:
             self._ctrlPin = Pin(ctrl_pin, mode=Pin.OUT)
         else:
             self._ctrlPin = None
 
-        self._t1char = (1000000 * (data_bits + stop_bits + 2)) // baudrate
+        # timing of 1 character in microseconds (us)
+        self._t1char = (1_000_000 * (data_bits + stop_bits + 2)) // baudrate
+
+        # inter-frame delay in microseconds (us)
+        # - <= 19200 bps: 3.5x timing of 1 character
+        # - > 19200 bps: 1750 us
         if baudrate <= 19200:
-            # 4010us (approx. 4ms) @ 9600 baud
-            self._t35chars = (3500000 * (data_bits + stop_bits + 2)) // baudrate
+            self._inter_frame_delay = (self._t1char * 3500) // 1000
         else:
-            self._t35chars = 1750   # 1750us (approx. 1.75ms)
+            self._inter_frame_delay = 1750
+
+        # no specific reason for 120, taken from _uart_read
+        # convert to us by multiplying by 1000
+        self._uart_read_timeout = (read_timeout or 0.120) * 1000
```

Comment:
I would like to have this in a dedicated PR as it is not related to the async functionality implementation. Additionally `0.120` feels a little bit mysterious :D (I agree the same mystery as with 40 repetitions)

`repetitions = self._uart_read_timeout // self._inter_frame_delay` has been 40 (something more than zero), this is no longer true with low baudrates, which is the default value.

```
_t1char = (1_000_000 * (8 + 1 + 2)) // 9600
# _t1char = 1145
_inter_frame_delay = (1145 * 3500) // 1000
# _inter_frame_delay = 4007
repetitions =  (0.120 * 1000) // 4007
# repetitions = 0
```

Reply:
The 0.120 was just to change it to 120 (the old value) as the default. But I agree, this should be its own PR - I'll revert these changes.